packages/integration-sdk-runtime/src/synchronization/index.ts

```diff
@@ -153,23 +163,35 @@ async function getPartialDatasets() {
 export async function uploadCollectedData(context: SynchronizationJobContext) {
   context.logger.synchronizationUploadStart(context.job);
 
-  await walkDirectory({
-    path: 'graph',
-    async iteratee({ filePath, data }) {
-      const parsedData = JSON.parse(data);
-
-      try {
-        if (Array.isArray(parsedData.entities)) {
-          await uploadData(context, 'entities', parsedData.entities);
-        }
-
-        if (Array.isArray(parsedData.relationships)) {
-          await uploadData(context, 'relationships', parsedData.relationships);
-        }
-      } catch (err) {
-        throw synchronizationApiError(err, 'Error uploading collected data');
-      }
-    },
+  await timeOperation({
+    logger: context.logger,
+    metricName: 'synchronization-upload-duration',
```

Comment:
`duration-sync-upload` to match the name of the `sync` command?

Reply:
Hum, perhaps just `duration-sync` to match `duration-collect`, and those are then `duration-<command>` 🤔

Reply:
hmm when I think of `duration-sync` I think of the duration of the entire synchronization process. I think I would prefer to stick with `duration-sync-upload` and have another metric  like `duration-sync-finalize` for tracking how long it takes our backend process to run. Later on we can figure out a good way to track the total duration (including the time waiting for sync to start) and name that `duration-sync`.

Reply:
Makes sense, thanks @charlieduong94